cmd/daprd/options/options.go

```diff
@@ -117,6 +118,7 @@ func New(args []string) *Options {
 	flag.IntVar(&opts.AppHealthProbeTimeout, "app-health-probe-timeout", int(config.AppHealthConfigDefaultProbeTimeout/time.Millisecond), "Timeout for app health probes in milliseconds")
 	flag.IntVar(&opts.AppHealthThreshold, "app-health-threshold", int(config.AppHealthConfigDefaultThreshold), "Number of consecutive failures for the app to be considered unhealthy")
 	flag.StringVar(&opts.AppChannelAddress, "app-channel-address", runtime.DefaultChannelAddress, "The network address the application listens on")
+	flag.StringVar(&opts.CanarySubscriber, "canary-subscriber", "", "Auto-generates a canary subscriber. use the pattern {PubSubName}:{TopicName}")
```

Comment:
I would like to find another way to create tell the runtime to create a canary subscriber which isn't a flag. Since this is a special string format anyway (Pubsub:Topic) can we not instead use the Subscription CRD and add an explicit, strongly typed `canary: true/false` field to it?

Reply:
@yaron2 thanks for taking the time to check out the PR!

> can we not instead use the Subscription CRD and add an explicit, strongly typed canary: true/false field to it?

This PR implementation already exactly does this (`canary: true` on the subscription yaml)

Additionally, I've also gone a step further and provided the CLI flag `canary-subscriber` which has the same effect as someone creating an entire subscription yaml and setting its values manually. It's a pure convenience, my thinking is this provides the most accessible and least error prone method of setting up a subscription which is ready to send a canary. But it is a convenience, so technically it can be removed. Happy to remove if you would prefer!

Reply:
It's best to remove the flag and make the configuration experience simple. We also generally prefer to keep CLI flags for global runtime configs and not for individual features as much as possible.

Reply:
Removed the convenience CLI flag.